src/tools/monmaptool.cc

```diff
@@ -230,7 +230,10 @@ int main(int argc, const char **argv)
       if (i == args.end())
 	helpful_exit();
       entity_addr_t addr;
-      if (!addr.parse(*i)) {
+      const char* endcompare = *i + strlen(*i);
```

Comment:
i'd suggest create a tiny helper just like `entity_name_t::parse()`, so it can be reused. but that helper should accept a `std::string_view` instead of `std::string` so it can be used without creating temporary `string` objects under some circumstances.

Reply:
something like:

```c++
bool entity_addr_t::parse(std::string_view s)
{
  const char* start = s.data();
  char *end = nullptr;
  bool got = parse(start, &end);
  return got && end == start + s.size();
}
```
and
```c++
if (!addr.parse(string_view{*i})) {
```